src/main/kotlin/net/maizegenetics/phgv2/cli/CreateRanges.kt

```diff
@@ -53,68 +65,74 @@ class CreateRanges: CliktCommand(help="Create BED file of reference ranges from
      * @param boundary A type of boundary. Either `gene` or `cds`
      * @param pad Number of base-pairs to flank boundary regions
      *
-     * @return A list of type `Pair<Int, Int>`
+     * Initially creates a `Triple<Position, Position, String>`
+     * The "String" in the Triple is a concatenated list of Gene ID.  Will be 1 or more,
+     * based on whethere overlapping genes were merged.
+     *
+     * This Triple is further processed to create a RangeMap<Position, String>
+     * of merged genes:  embedded genes are tossed, overlapping genes are merged.
      */
-    fun idMinMaxBounds(genes: List<Feature>, boundary: String, pad: Int): List<Pair<Int, Int>> {
+    fun idMinMaxBounds(refSeq: Map<String, NucSeq>, genes: List<Feature>, boundary: String): RangeMap<Position, String> {
         val boundMinMax = when (boundary) {
             "gene" -> genes.map {gene ->
-                Pair(gene.start, gene.end)
+                Triple(Position(gene.seqid,gene.start), Position(gene.seqid,gene.end), gene.attribute("ID").first())
             }
             "cds" -> {
```

Comment:
Should these possibilities be an enum?

Reply:
I left the code as it was initially written - if the consensus of the programmers is this should be an enum I can change the code.